src/parser.ts

```diff
@@ -0,0 +1,395 @@
+import { converters } from './converters';
+import { RQLParseError } from './errors';
+
+/**
+ * An interface for RQL parsed into nested object form
+ *
+ * @export
+ * @interface RQLOperator
+ */
+export interface RQLOperator {
+  name: string;
+  args: Array<RQLOperator | any>;
+}
+
+/**
+ * User-Defined Type Guard for the RQLOperator interface
+ *
+ * @export
+ * @param arg an unknown value
+ * @returns the arg cast as RQLOperator
+ */
+export function isRQLOperator(arg: any): arg is RQLOperator {
+  return arg && arg.name && typeof arg.args === 'object' && Array.isArray(arg.args);
```

Comment:
`typeof arg.args === 'object'` isn't needed. `Array.isArray()` should handle that part for you

Reply:
fixed